scripts/gen_packaging_stats.pl

```diff
@@ -0,0 +1,220 @@
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+=head1 NAME
+
+gen_packaging_stats.pl - Generates aggregated data about the packaging components of products for a specific category in a specific country
+
+=head1 DESCRIPTION
+
+Aggregation counts are stored in a structure of the form:
+
+{
+    countries => {
+        "en:world" => ..
+        "en:france" => {
+            categories => {
+                "all" => .. # stats for all categories
+                "en:yogourts" => {
+                    shapes => {
+                        "en:unknown" => ..
+                        "all" => .. # stats for all shapes
+                        "en:bottle" => {
+                            materials_parents => .. # stats for parents materials (e.g. PET will also count for plastic)
+                            materials => {
+                                "all" => ..
+                                "en:plastic" => 12, # number of products sold in France that are yogurts and that have a plastic bottle packaging component
+                            }
+                        },
+                        ..
+                    }
+                },
+                ..
+            }
+        },
+        ..
+    }
+}
+
+=cut
+
+use ProductOpener::PerlStandards;
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Store qw/:all/;
+use ProductOpener::Tags qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Lang qw/:all/;
+use ProductOpener::Data qw/:all/;
+
+use File::Path qw(mkpath);
+use JSON::PP;
+use Data::DeepAccess qw(deep_exists deep_get deep_set deep_val);
+
+=head2 generate_packaging_stats_for_query($name, $query_ref)
+
+Generate packaging stats for products matching a specific query.
+
+Stats are saved in .sto format in $data_root/data/categories_stats/
+and in JSON format in $www_root/data/categories_stats/
+
+=head3 Arguments
+
+=head4 name $name
+
+
+
+=head4 query reference $query_ref
+
+=cut
+
+sub generate_packaging_stats_for_query ($name, $query_ref) {
+
+	$query_ref->{'empty'} = {"\$ne" => 1};
+	$query_ref->{'obsolete'} = {"\$ne" => 1};
+
+	my $fields_ref = {
+		countries_tags => 1,
+		categories_tags => 1,
+		packagings => 1,
+	};
+
+	my $socket_timeout_ms = 3 * 60 * 60 * 60000;    # 3 hours
+	my $products_collection = get_products_collection($socket_timeout_ms);
+
+	my $products_count = $products_collection->count_documents($query_ref);
+
+	print STDERR "$name: $products_count products\n";
```

Comment:
If we are going to put that in a cron, it would be better to have an option to disable "informative" prints and only keeps real errors, to have mail only on real failures.

Reply:
Agreed, I will add a --quiet option